index.bs

```diff
@@ -2094,6 +2166,40 @@ operation has no [=authenticator attachment modality=] selection option, so the
 credential|credentials=]. The [=client=] and user will then use whichever is available and convenient at the time.
 
 
+### Resident Key Requirement Enumeration (enum <dfn enum>ResidentKeyRequirement</dfn>) ### {#residentKeyRequirement}
+
+<xmp class="idl">
+    enum ResidentKeyRequirement {
+        "discouraged",
+        "preferred",
+        "required"
+    };
+</xmp>
+
+This enumeration's values describe the [=[RP]=]'s requirements for [=resident credentials=]:
+
+<div dfn-type="enum-value" dfn-for="ResidentKeyRequirement">
+    :   <dfn>discouraged</dfn>
+    ::  This value indicates the client SHOULD try to use an authenticator capable of a [=non-resident credential=] 
```

Comment:
(This wording is also based around storage of private keys, which the platform doesn't generally have insight into, vs whether a credential can be returned with an empty allowList. It's also based around authenticator selection but I don't think we would implement that way vs using this information to make a decision about the rk flag after the user has selected an authenticator.)

Reply:
I have changed the description for both discouraged and preferred to better convey the intent of client/authenticator behaviour. Will let the next round of review decide if that's ok.